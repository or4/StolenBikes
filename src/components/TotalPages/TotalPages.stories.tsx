import * as React from 'react';
import { storiesOf } from '@storybook/react';

import { TotalPages } from './TotalPages';

const stories = storiesOf('TotalPages', module);

stories.add('TotalPages', () => {
    return <TotalPages totalPages={18} />;
});
