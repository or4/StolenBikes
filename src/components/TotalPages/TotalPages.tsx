import React from 'react';
import styled from 'styled-components';

export interface IProps {
    totalPages: number;
}

const Container = styled.div`
    min-height: 40px;
    line-height: 40px;
    text-align: right;
`;

export const TotalPages: React.FC<IProps> = ({ totalPages }: IProps) => <Container>{`Total: ${totalPages}`}</Container>;
