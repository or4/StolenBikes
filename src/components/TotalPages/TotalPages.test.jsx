import React from 'react';
import { mount } from 'enzyme';

import { TotalPages } from './TotalPages';

describe('TotalPages', () => {
    it('should render correctly', () => {
        const component = <TotalPages />;

        const container = mount(component);
        expect(container).toMatchSnapshot();
        container.unmount();
    });
});
