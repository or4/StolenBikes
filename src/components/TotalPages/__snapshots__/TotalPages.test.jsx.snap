// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalPages should render correctly 1`] = `
<TotalPages>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "jaMoQF",
            "rules": Array [
              "
    min-height: 40px;
    line-height: 40px;
    text-align: right;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa jaMoQF"
      >
        Total: undefined
      </div>
    </StyledComponent>
  </styled.div>
</TotalPages>
`;
